modules/simple_forms_api/app/services/simple_forms_api/pdf_stamper.rb

```diff
@@ -230,11 +229,23 @@ def self.stamp_submission_date(stamped_template_path, config)
         page_configuration = default_page_configuration
         page_configuration[config[:page_number]] = { type: :text, position: date_text_stamp_position }
 
-        multistamp(stamped_template_path, Time.current.in_time_zone('UTC').strftime('%H:%M %Z %D'), page_configuration,
-                   12)
+        current_time = Time.current.in_time_zone('UTC').strftime('%H:%M %Z %D')
+        verified_stamp(stamped_template_path) do
+          multistamp(stamped_template_path, current_time, page_configuration, 12)
+        end
       end
     end
 
+    def self.verified_stamp(template_path)
+      orig_size = File.size(template_path)
+      yield
```

Comment:
For the uninitiated: They ruby `yield` command will run whatever block is passed into the method, in this instance that would be the stamp/multistamp methods.